utils/blockchain.py

```diff
@@ -577,3 +579,17 @@ def wait_until_transaction_hash_exists_in_blockchain(tx_hash, network_id):
             break
 
         i += 1
+
+
+def is_valid_ethereum_address(address: str) -> bool:
```

Comment:
Just use `Web3.is_address("0xBadAddress")`
Read more here: https://web3py.readthedocs.io/en/stable/web3.main.html#web3.Web3.is_address

If the correct address format is also important - use `Web3.is_checksum_address("0xNonChecksumAddress")` validation
https://web3py.readthedocs.io/en/stable/web3.main.html#web3.Web3.is_checksum_address

And if you need to convert to the correct format then: `Web3.to_checksum_address("0xNonChecksumAddress")`
https://web3py.readthedocs.io/en/stable/web3.main.html#web3.Web3.to_checksum_address

Reply:
Thanks! Changed the validation to this built-in function